improver/utilities/field_texture.py

```diff
@@ -0,0 +1,173 @@
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Module containing plugin to calculate the texture in a field considering edge transitions."""
+
+import iris
+import numpy as np
+from iris.exceptions import CoordinateNotFoundError
+
+from improver import BasePlugin
+from improver.metadata.constants.attributes import MANDATORY_ATTRIBUTE_DEFAULTS
+from improver.metadata.utilities import create_new_diagnostic_cube
+from improver.nbhood.square_kernel import SquareNeighbourhood
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_manipulation import collapsed
+
+
+class FieldTexture(BasePlugin):
+    """Plugin to calculate the texture in a binary field considering edge transitions
+
+    Code methodology:
+
+    1) Takes a binary field that has been thresholded at a cloud cover fraction
+       of interest and looks for the transitions/edges in the field that mark
+       out a transition from cloudy to non-cloudy grid cell.
+    2) The transition calculation is then fed into the neighbourhooding code
+       (_calculate_ratio) to calculate a ratio for each cell.
+    3) The new cube of ratios is then thresholded and the realization coordinates
+       are collapsed to generate a mean of the thresholded ratios.
+    """
+
+    def __init__(self, nbhood_radius, ratio_threshold):
+        """
+        The class is called with the default constraints for the processing code.
```

Comment:
To keep this generic, I think it would be better to set the default values in the CLI.
For cloud we are using 0.05 but I guess for precip it would be different? 

Removed that line.